.github/workflows/goapi.image+upload.yaml

```diff
@@ -0,0 +1,90 @@
+name: GO API Deploy to Amazon ECR
```

Comment:
@nick-bisonai Please include the following checks:

* https://pkg.go.dev/cmd/vet (`go vet`)
* [shadow](https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow) (`go vet -vettool=$(which shadow)`)